tools/c7n_mailer/c7n_mailer/azure/deploy.py

```diff
@@ -87,22 +88,24 @@ def provision(config):
                    entry_point=os.path.join(os.path.dirname(__file__), 'handle.py'),
                    extra_modules={'c7n_mailer', 'ruamel'})
 
-    packager.pkg.add_contents(
-        function_name + '/config.json',
-        contents=json.dumps(config))
-
     packager.pkg.add_contents(
         function_name + '/function.json',
         contents=packager.get_function_config({'mode':
                                               {'type': 'azure-periodic',
                                                'schedule': schedule}}))
     # Add mail templates
-    template_dir = os.path.abspath(
-        os.path.join(os.path.dirname(__file__), '../..', 'msg-templates'))
-
-    for t in os.listdir(template_dir):
-        with open(os.path.join(template_dir, t)) as fh:
-            packager.pkg.add_contents('msg-templates/%s' % t, fh.read())
+    for d in config['templates_folders']:
+        if not os.path.exists(d):
+            continue
+        for t in [f for f in os.listdir(d) if os.path.splitext(f)[1] == '.j2']:
+            with open(os.path.join(d, t)) as fh:
+                packager.pkg.add_contents(function_name + '/msg-templates/%s' % t, fh.read())
```

Comment:
nit: typically its nice to use os.path.join but its given its always linux in this context prolly irrelevant.

Reply:
If we fix functions deployment from Windows, os.path.join won't work, right?

Reply:
This is referencing a local path right?  So as-is would fail in windows.  I saw a bunch of hard coded separators but they were there before you made your changes too so I didn't worry about it all being fixed here.